integration-tests/src/test_loop/tests/chunk_validator_kickout.rs

```diff
@@ -0,0 +1,115 @@
+        // Target giving one mandate to each chunk validator, which results in
+        // every chunk validator validating only one shard in most cases.
+        .target_validator_mandates_per_shard(1);
+    for account in &accounts {
+        genesis_builder.add_user_account_simple(account.clone(), initial_balance);
+    }
+    let genesis = genesis_builder.build();
+
+    let TestLoopEnv { mut test_loop, datas: node_datas } = builder
+        .genesis(genesis)
+        .clients(clients)
+        // Drop only chunks validated by `VALIDATOR_TO_KICKOUT`.
+        // By how our endorsement stats are computed, this will count as this
+        // validator validating zero chunks.
+        .drop_chunks_validated_by(account_id)
+        .build();
+
+    // Run chain until our targeted chunk validator is (not) kicked out.
+    let client_handle = node_datas[0].client_sender.actor_handle();
+    let initial_validators = get_epoch_all_validators(&test_loop.data.get(&client_handle).client);
+    assert_eq!(initial_validators.len(), 8);
+    assert!(initial_validators.contains(&account_id.to_string()));
+    let success_condition = |test_loop_data: &mut TestLoopData| -> bool {
+        let client = &test_loop_data.get(&client_handle).client;
+        let validators = get_epoch_all_validators(client);
+        let tip = client.chain.head().unwrap();
+        let epoch_height =
+            client.epoch_manager.get_epoch_height_from_prev_block(&tip.prev_block_hash).unwrap();
+
+        if expect_kickout {
+            assert!(epoch_height < 4);
+            return if validators.len() == 7 {
+                assert!(!validators.contains(&account_id.to_string()));
+                true
+            } else {
+                false
+            };
+        } else {
+            assert_eq!(validators.len(), 8, "No kickouts are expected");
+            epoch_height >= 4
+        }
+    };
+
+    test_loop.run_until(
+        success_condition,
+        // Timeout at producing 5 epochs, approximately.
+        Duration::seconds((5 * epoch_length) as i64),
+    );
+
+    TestLoopEnv { test_loop, datas: node_datas }
+        .shutdown_and_drain_remaining_events(Duration::seconds(20));
+}
+
+/// Get all validator account names for the latest epoch.
+fn get_epoch_all_validators(client: &Client) -> Vec<String> {
+    let tip = client.chain.head().unwrap();
+    let epoch_id = tip.epoch_id;
+    let all_validators = client.epoch_manager.get_epoch_all_validators(&epoch_id).unwrap();
+    all_validators.into_iter().map(|vs| vs.account_id().to_string()).collect()
+}
+
+/// Checks that chunk validator with low endorsement stats is kicked out.
+#[test]
+fn test_chunk_validator_kicked_out() {
+    run_test_chunk_validator_kickout(VALIDATOR_TO_KICKOUT, true);
+}
+
+/// Checks that block producer with low chunk endorsement stats is not kicked out.
+#[test]
+fn test_block_producer_not_kicked_out() {
+    run_test_chunk_validator_kickout(VALIDATOR_TO_AVOID_KICKOUT, false);
```

Comment:
I do not get this, how is account3 different from account6? is it because account3 is producer and account6 is validator? can we make it more explicit in comments or check with assertions?